test/gleam/bit_array_test.gleam

```diff
@@ -308,3 +308,54 @@ pub fn inspect_partial_bytes_test() {
   bit_array.inspect(<<5:3, 11:4, 1:2>>)
   |> should.equal("<<182, 1:size(1)>>")
 }
+
+pub fn index_of_found_test() {
+  <<"Hello, World":utf8>>
+  |> bit_array.index_of(<<", ":utf8>>)
+  |> should.equal(40)
+}
+
+pub fn index_of_not_found_test() {
+  <<"Hello, World":utf8>>
+  |> bit_array.index_of(<<"Joe":utf8>>)
+  |> should.equal(-1)
+}
+
+pub fn split_once_found_test() {
+  <<"Hello, World":utf8>>
+  |> bit_array.split_once(<<", ":utf8>>)
+  |> should.be_ok
+  |> should.equal(#(<<"Hello":utf8>>, <<"World":utf8>>))
+}
+
+pub fn split_once_empty_needle_test() {
+  <<"Hello, World":utf8>>
+  |> bit_array.split_once(<<>>)
+  |> should.be_ok
+  |> should.equal(#(<<>>, <<"Hello, World":utf8>>))
+}
+
+pub fn split_once_not_found_test() {
+  <<"Hello, World":utf8>>
+  |> bit_array.split_once(<<"Joe":utf8>>)
+  |> should.be_error
+}
+
+@target(erlang)
+pub fn split_once_non_byte_aligned_found_test() {
+  // haystack: 00000001 00000010 00000011 00000100 00000101
+  // needle:                           11
+  // before:   00000001 00000010 000000
+  // after:                               00000100 00000101
+  <<1, 2, 3, 4, 5>>
+  |> bit_array.split_once(<<3:2>>)
+  |> should.be_ok
+  |> should.equal(#(<<1, 2, 0:6>>, <<4, 5>>))
+}
```

Comment:
What a great test! Very clear! Thank you

Reply:
thanks